modules/base/rendering/screenspacetimevaryingimageonline.cpp

```diff
@@ -0,0 +1,242 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/rendering/screenspacetimevaryingimageonline.h>
+
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/json.h>
+#include <openspace/util/timemanager.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/io/texture/texturereader.h>
+#include <ghoul/logging/logmanager.h>
+#include <ghoul/opengl/texture.h>
+#include <fstream>
+
+namespace {
+    constexpr std::string_view _loggerCat = "ScreenSpaceTimeVaryingImageOnline";
+
+    constexpr openspace::properties::Property::PropertyInfo FileInfo = {
+        "FilePath",
+        "File Path",
+        "The file path to the data containing information about when to display which image.",
+        openspace::properties::Property::Visibility::User
+    };
+
+    // This `ScreenSpaceRenderable` displays an image based on the current in-game
+    // simulation time. The image shown is selected from a JSON file containing
+    // timestamp-URL pairs. The image with the closest timestamp before or equal to the
+    // current time is displayed.
+    // Example JSON format:
+    // {
+    //   "files": [
+    //     {"timestamp": "2024-05-10 15:00:00.0","url": "https://example.com/image1.png"},
+    //     {"timestamp": "2024-05-10 15:02:00.0","url": "https://example.com/image2.png"}
+    //   ]
+    // }
+    struct [[codegen::Dictionary(ScreenSpaceTimeVaryingImageOnline)]] Parameters {
+        // [[codegen::verbatim(FileInfo.description)]]
+        std::filesystem::path filePath;
+    };
+#include "screenspacetimevaryingimageonline_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation ScreenSpaceTimeVaryingImageOnline::Documentation() {
+    return codegen::doc<Parameters>("base_screenspace_time_varying_image_online");
+}
+
+ScreenSpaceTimeVaryingImageOnline::ScreenSpaceTimeVaryingImageOnline(
+                                                      const ghoul::Dictionary& dictionary)
+    : ScreenSpaceRenderable(dictionary)
+    , _jsonFilePath(FileInfo, "")
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _jsonFilePath = p.filePath.string();
+    addProperty(_jsonFilePath);
+
+    _jsonFilePath.onChange([this]() {
+        loadJsonData(_jsonFilePath.value());
+    });
```

Comment:
```suggestion
    _jsonFilePath.onChange([this]() {
        loadJsonData(_jsonFilePath.value());
    });
    addProperty(_jsonFilePath);
```

Didn't see this before either; we usually set up all of the properties and then "finalize" it with the `addProperty`. Doesn't make a real difference, but to keep the style the same